packages/superset-ui-preset-chart-xy/src/encodeable/ChannelEncoder.ts

```diff
@@ -24,9 +24,9 @@ export default class ChannelEncoder<Def extends ChannelDef<Output>, Output exten
   readonly definition: Def;
   readonly options: ChannelOptions;
 
-  protected readonly getValue: (datum: PlainObject) => Value | undefined;
-  readonly encodeValue: (value: any) => Output | null | undefined;
-  readonly formatValue: (value: any) => string;
+  protected readonly getValue: (datum: PlainObject) => ChannelInput;
+  readonly encodeValue: (value: ChannelInput) => Output | null | undefined;
+  readonly formatValue: (value: ChannelInput | { toString(): string }) => string;
```

Comment:
hmm, what is the `toString` for?

Reply:
One of the formatter function's minimum requirement is something that can be converted into a `string`.